internal/dataplane/parser/translate_grpcroute.go

```diff
@@ -0,0 +1,133 @@
+package parser
+
+import (
+	"fmt"
+
+	"github.com/kong/go-kong/kong"
+	gatewayv1alpha2 "sigs.k8s.io/gateway-api/apis/v1alpha2"
+	gatewayv1beta1 "sigs.k8s.io/gateway-api/apis/v1beta1"
+
+	"github.com/kong/kubernetes-ingress-controller/v2/internal/dataplane/kongstate"
+	"github.com/kong/kubernetes-ingress-controller/v2/internal/util"
+)
+
+// -----------------------------------------------------------------------------
+// Translate GRPCRoute - IngressRules Translation
+// -----------------------------------------------------------------------------
+
+// ingressRulesFromGRPCRoutes processes a list of GRPCRoute objects and translates
+// then into Kong configuration objects.
+func (p *Parser) ingressRulesFromGRPCRoutes() ingressRules {
```

Comment:
WDYT about adding some Parser unit tests that would cover GRPCRoute translations?

Reply:
While writing this unit test I realized there Is a lot of code to write (the test and all the helpers, if we want to keep consistency with the HTTPRoute approach). Since I think this PR is huge enough, and the basic functionality is already covered by the examples integration tests, I think we can leave it as it is. I'd like to open a new issue to improve the GRPC integration tests with more scenario, and we can include these unit tests there in that issue, so that we don't lose track of your suggestion. Does it make sense to you @czeslavo?